snakemake: 9.5.1 -> 9.11.8

kyehn opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
https://github.com/snakemake/snakemake/releases/tag/v9.11.8
https://github.com/snakemake/snakemake/blob/v9.11.8/CHANGELOG.md
https://github.com/snakemake/snakemake/compare/v9.5.1...v9.11.8

https://github.com/snakemake/snakemake-interface-common/releases/tag/v1.22.0
https://github.com/snakemake/snakemake-interface-common/compare/v1.17.4...v1.22.0

Snakemake Scheduler Plugin Interface: This package provides a stable interface for interactions between Snakemake and its scheduler plugins. https://github.com/snakemake/snakemake-interface-scheduler-plugins


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kyehn committed:
snakemake: 9.5.1 -> 9.11.6

Changelog: https://github.com/snakemake/snakemake/blob/v9.11.6/CHANGELOG.md

kyehn committed:
python313Packages.snakemake-interface-common: 1.17.4 -> 1.21.0

Changelog: https://github.com/snakemake/snakemake-interface-common/releases/tag/v1.21.0

kyehn committed:
python313Packages.snakemake-interface-scheduler-plugins: init at 2.0.1

veprbl commented on pkgs/development/python-modules/snakemake-interface-scheduler-plugins/default.nix:
```diff
@@ -0,0 +1,42 @@
+  # enabledTestPaths = [ "tests/tests.py" ];
+
+  meta = {
+    description = "Common functions and classes for Snakemake and its plugins";
```
```suggestion
    description = "Provides a stable interface for interactions between Snakemake and its scheduler plugins";
```

kyehn committed:
snakemake: 9.11.6 -> 9.11.8

Changelog: https://github.com/snakemake/snakemake/blob/v9.11.8/CHANGELOG.md

kyehn committed:
python313Packages.snakemake-interface-common: 1.21.0 -> 1.22.0

Changelog: https://github.com/snakemake/snakemake-interface-common/releases/tag/v1.22.0
